### PR TITLE
@eessex => Tag Admin

### PR DESCRIPTION
**Pull request by kanaabe**
tagadmin
tagadminerror

### CODE REVIEW

File: api/apps/tags/model.coffee [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+@destroy = (id, callback) ->
+  db.tags.remove { _id: ObjectId(id) }, callback
+
+#
+# JSON views
+#
+@present = (tag) =>
+  _.extend
+    id: tag?._id?.toString()
+  , _.omit(tag, '_id')
```
Comment (kanaabe) [inline api/apps/tags/model.coffee]: Decided to go with making a proper API endpoint for tags. This uses a `public` field to determine if it's a topic tag or internal tag. The association on the article side is still a question, but for now this makes CRUD-ing tags easier.

File: client/apps/articles_list/client/client.coffee [hunk 2]
```diff
@@ -10,7 +10,7 @@ sd = require('sharify').data
 request = require 'superagent'
 icons = -> require('../icons.jade') arguments...
 
-module.exports = ArticlesListView = React.createClass
+module.exports.ArticlesListView = ArticlesListView = React.createClass
```
Comment (kanaabe) [inline client/apps/articles_list/client/client.coffee]: Not quite sure why these were removed @eessex but I added them back into some components. This makes testing a bit easier -- 

```
{ ArticlesListView } = mod = require '... 
mod.__set__ 'SomeOtherComponent', sinon.stub()
React.createElement(ArticlesListView, props)
```
That way we can pass in ArticlesListView to the React rendering side AND modify the requires on the file. Lmk if that makes sense.

Comment (eessex) [inline client/apps/articles_list/client/client.coffee]: Ah cool -- I think they were removed when the displayName/new react naming was added to them, but didn't realize they added this utility.

Comment (kanaabe) [inline client/apps/articles_list/client/client.coffee]: 👍  nw, also sorry this PR is giant. Might help to look per commit

File: client/apps/articles_list/client/client.coffee [hunk 3]
```diff
@@ -66,9 +66,10 @@ module.exports = ArticlesListView = React.createClass
         FilterSearch {
           url: sd.API_URL + "/articles?published=#{@state.published}&channel_id=#{sd.CURRENT_CHANNEL.id}&q=%QUERY"
           placeholder: 'Search Articles...'
-          articles: @state.articles
+          collection: @state.articles
           searchResults: @setResults
           selected: null
+          contentType: 'article'
         }
```
Comment (eessex) [inline client/apps/articles_list/client/client.coffee]: Nice that this is becoming cross-type!

File: client/apps/settings/client/tags.coffee [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+          public: isPublic
+
+  getActiveState: (type) ->
+    if @state.public is type then 'is-active' else ''
+
+  render: ->
+    div {
+      className: 'tags-container'
+      'data-loading': @state.loading
+      ref: 'tagsContainer'
```
Comment (eessex) [inline client/apps/settings/client/tags.coffee]: Looks like this is only used for testing, you can locate using the classname too.

File: client/components/tag_list/test/index.coffee [hunk 5]
```diff
@@ -0,0 +1,76 @@
+benv = require 'benv'
+sinon = require 'sinon'
+{ resolve } = require 'path'
+fixtures = require '../../../../test/helpers/fixtures'
+React = require 'react'
+ReactDOM = require 'react-dom'
+ReactTestUtils = require 'react-addons-test-utils'
+ReactDOMServer = require 'react-dom/server'
+r =
+  findAll: ReactTestUtils.scryRenderedDOMComponentsWithClass
```
Comment (eessex) [inline client/components/tag_list/test/index.coffee]: scryRenderedDOMComponentsWithClass is an exciting new one for me

File: client/apps/settings/client/add_tag.coffee [hunk 6]
```diff
@@ -0,0 +1,22 @@
+React = require 'react'
+ReactDOM = require 'react-dom'
+{ div, input, button } = React.DOM
+
+module.exports = AddTag = React.createClass
+  displayName: 'TagsView'
+
+  addTag: ->
+    @props.addTag @refs.addTagInput.value
+    @refs.addTagInput.value = ''
```
Comment (eessex) [inline client/apps/settings/client/add_tag.coffee]: I believe using state is the recommended method over refs - this is so concise that it doesn't seem pressing to give a state to this form.
https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs

**Comment by eessex** [comment]
A few small comments but nothing that would hold up merging, excited to get this in action!

File: client/apps/settings/client/add_tag.coffee (hunk 6 above)
Comment (kanaabe) [inline client/apps/settings/client/add_tag.coffee]: Good catch! Changes made, will merge when 💚

**Pull request merged**